dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -0,0 +1,124 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2020 Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.monitor
+
+import com.datadog.android.core.internal.data.Writer
+import com.datadog.android.core.internal.time.TimeProvider
+import com.datadog.android.core.internal.utils.devLogger
+import com.datadog.android.rum.RumContext
+import com.datadog.android.rum.RumMonitor
+import com.datadog.android.rum.internal.domain.RumEvent
+import com.datadog.android.rum.internal.domain.RumEventData
+import java.util.UUID
+
+internal class DatadogRumMonitor(
+    applicationId: UUID,
+    private val timeProvider: TimeProvider,
+    private val writer: Writer<RumEvent>
+) : RumMonitor {
+
+    private var context: RumContext =
+        RumContext(
+            applicationId,
+            UUID.randomUUID(),
+            null
+        )
+
+    private val activeView = mutableMapOf<Any, RumEvent>()
+
+    // region RumMonitor
+
+    override fun addUserAction(
+        action: String,
+        attributes: Map<String, Any?>
+    ) {
+        val eventData = RumEventData.UserAction(action)
+        val event = RumEvent(
+            context,
+            timeProvider.getServerTimestamp(),
+            eventData,
+            attributes
+        )
+        writer.write(event)
+    }
+
+    override fun startView(
+        key: Any,
+        className: String,
+        attributes: Map<String, Any?>
+    ) {
+        val activeViewId = UUID.randomUUID()
```

Comment:
if we do it like this means that for same Activity view you will have different viewIds each time. Means that If I perform a TAP on that activity, switch to another, then switch back to the same activity and perform a new TAP those 2 taps will not count as 2 TAPs on the same Page but as 1 separated TAP on a different Page. Is that what what we want ?

Reply:
Yes, it'll appear as 2 different views, each one with their own tap.
We defined a view as being what happens between `onResume` and `onPause` and we can't know what is the navigation experience while the activity is not in foreground. 
Besides, because the activity could be killed by the OS  and restored when the user navigates back to it, it's better to keep consistency. Whether or not the os kills the activity, the result will always be the same.